.github/workflows/pre-commit-autoupdate.yml

```diff
@@ -0,0 +1,79 @@
+name: "Run pre-commit autoupdate"
+
+on:
+  workflow_call:
+    inputs:
+      target:
+        description: "The branch to issue the PR against"
+        type: "string"
+        default: "main"
+    secrets:
+      TOKEN:
+        description: "Your token"
+        required: true
+
+permissions:
+  contents: write
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  precommit-autoupdate:
+    name: "pre-commit autoupdate"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check out repository"
+        uses: actions/checkout@v4
+
+      - name: "Create a new branch"
+        id: branch
+        run: |
+          branch=pre-commit/$(date +'%Y-%m-%dT%H_%M_%S')
+          git checkout -b $branch
+          echo "name=$branch" >> $GITHUB_OUTPUT
+
+      - name: "Add brew to the PATH"
+        run: echo "/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin" >> $GITHUB_PATH
+
+      - name: "Install pre-commit"
+        run: brew install pre-commit
+
+      - name: "Check for pre-commit hook updates"
+        id: pre-commit
+        run: |
+          pre-commit autoupdate
+          updates=$(git diff --raw)
+          echo "updates=$updates" >> $GITHUB_OUTPUT
+
+      - name: "Commit local changes"
+        if: ${{ steps.pre-commit.outputs.updates }}
+        run: |
+          git config user.name "$USER"
+          git config user.email "$EMAIL"
+          git status
+          git add .
+          git commit -m "$MESSAGE"
+        env:
+          USER: "Github Build Bot"
+          EMAIL: "buildbot@fishtownanalytics.com"
+          MESSAGE: "[automated] update pre-commit hooks"
+
+      - name: "Open pull request"
```

Comment:
Won't this fail when there are no updates because there are no commits?

Reply:
Oh, I thought if the previous step was skipped, then the remaining steps got skipped. I check the results of `git diff` in the previous to see if I have anything to commit. Do I need to add that conditional (or something like `if: success()` to the following steps too?

Reply:
success() evaluates whether all previous steps that actually ran were successful. It does not take into account steps that were skipped due to conditions. If the previous step was skipped and all other preceding steps ran successfully, then success() will evaluate to true, and this will still run.  You need to check the same condition you're checking for if there was a diff.